sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationTokenCredential.java

```diff
@@ -177,7 +207,12 @@ private void setToken(String freshTokenString) {
             public void run() {
                 try {
                     Mono<String> tokenAsync = tokenCache.fetchFreshToken();
-                    tokenCache.setToken(tokenAsync.block());
+                    String freshTokenString = tokenAsync.block();
```

Comment:
Is blocking necessary here? Would subscribing to the mono and handling fresh token there be more "reactive"? Something like
```
tokenAsync.subscribe(token -> {
   if (!tokenCache.isTokenExpired(token)) {
      tokenCache.setToken(token);
    } else {
       logger.logExceptionAsError(new IllegalArgumentException("The token returned from the tokenRefresher is expired."));
 }
});
```